test/h5test.h

```diff
@@ -332,126 +288,1045 @@ extern "C" {
-H5TEST_DLLVAR  H5_ATOMIC(size_t) n_tests_run_g;
-H5TEST_DLLVAR  H5_ATOMIC(size_t) n_tests_passed_g;
-H5TEST_DLLVAR  H5_ATOMIC(size_t) n_tests_failed_g;
-H5TEST_DLLVAR  H5_ATOMIC(size_t) n_tests_skipped_g;
-
-H5TEST_DLLVAR uint64_t vol_cap_flags_g;
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
+ *
+ * \brief "Sends" a message to another testing process using a temporary file
+ *
+ * \param[in]  file Name of the temporary file to be created
+ * \param[in]  arg1 First argument to be written to the temporary file
+ * \param[in]  arg2 Second argument to be written to the temporary file
+ *
+ * \return nothing
+ *
+ * \details h5_send_message() facilitates inter-process communication in
+ *          tests by "sending" a message to another process using a temporary
+ *          file. \p file is the name of the temporary file to be created.
+ *          \p arg1 and \p arg2 are strings to be written to the first and
+ *          second lines of the file, respectively, and may both be NULL.
+ *
+ *          When there are multiple test processes that need to communicate
+ *          with each other, they do so by writing and reading signal files
+ *          on disk, the names and contents of which are used to inform a
+ *          process about when it can proceed and what it should do next.
+ *
+ * \see h5_wait_message()
+ *
+ */
+H5TEST_DLL void h5_send_message(const char *file, const char *arg1, const char *arg2);
 
-H5TEST_DLL void   h5_send_message(const char *file, const char *arg1, const char *arg2);
+/**
+ * --------------------------------------------------------------------------
+ * \ingroup H5TEST
+ *
+ * \brief Waits for a message from another testing process to be available
+ *
+ * \param[in]  file Name of the temporary file to read
+ *
+ * \return \herr_t
+ *
+ * \details h5_wait_message() facilitates inter-process communication in
+ *          tests by waiting until a temporary file written by the
+ *          h5_send_message() function is available for reading. \p file is
+ *          the name of the file being waited on and should match the
+ *          filename provided to h5_send_message(). This function repeatedly
+ *          attempts to open a file with the given filename until it is
+ *          either successful or times out. The temporary file is removed
+ *          once it has been successfully opened.
+ *
+ *          When there are multiple test processes that need to communicate
+ *          with each other, they do so by writing and reading signal files
+ *          on disk, the names and contents of which are used to inform a
+ *          process about when it can proceed and what it should do next.
+ *
+ * \see h5_send_message()
+ *
+ */
 H5TEST_DLL herr_t h5_wait_message(const char *file);
 
+/* Generally useful testing routines */
+H5TEST_DLL void h5_clean_files(const char *base_name[], hid_t fapl);
+H5TEST_DLL void h5_reset(void);
+
+/* TODO: documentation */
```

Comment:
Similar to the above, documentation should likely be added at some point for `generate_threadlocal_filename` if it stays here, but I think the function should be moved out of h5test.h after considering where it's actually useful. With this PR, there's a testframe.h header for things that should only exist in that testing framework. I moved some things around to try and keep the two testing frameworks separate, but more work is needed.